### PR TITLE
Adding support for searching project by a variety of keywords

### PR DESCRIPTION
**Pull request by jeyrschabu**
- Added support for a fuzzy string match on object attributes
- Added support for pageSize similar to applications
- Search is case insensitive on key and value and allows searching even deep attributes on the object
- Using the same ranking algo for search result relevance as applications (Worth to note that this doesnt apply to deep properties)

### CODE REVIEW

**Comment by jeyrschabu** [comment]
@ajordens PTAL

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ProjectsController.groovy [hunk 1]
```diff
@@ -72,10 +73,14 @@ public class ProjectsController {
     return projects
   }
 
-  @ApiOperation(value = "", notes = "Fetch all projects")
+  @ApiOperation(value = "", notes = """Fetch all projects.Support filtering by one or more attributes:
```
Comment (ajordens) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ProjectsController.groovy]: Worth a newline between `projects.` and `Supports`

```
Fetch all projects.

Supports filtering by one or more attributes:
...
```

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ProjectsController.groovy [hunk 2]
```diff
@@ -122,6 +127,51 @@ public class ProjectsController {
...
+    items.sort { Project a, Project b ->
+      return SearchUtils.score(b, attributes) - SearchUtils.score(a, attributes)
+    }
+
+  }
+
+  def fuzzyPredicate = { Project project, String key, String value  ->
+    boolean matches = SearchUtils.matchesIgnoreCase(project.properties, key, value)
+    if (!matches) {
+      if (project.config.pipelineConfigs.find { SearchUtils.matchesIgnoreCase(it.properties, key, value) } ) {
```
Comment (ajordens) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ProjectsController.groovy]: I suspect that looking into `pipelineConfigs` is probably not providing any value

**Comment by jeyrschabu** [comment]
made suggested changes. merging down

**Pull request merged**